mainwindow.cpp

```diff
@@ -363,41 +326,22 @@ void MainWindow::openPasswordFile()
     if (dialog.exec()) {
         QStringList fileNames = dialog.selectedFiles();
         readPasswdFiles(fileNames);
+        m_ui->actionResumeAttack->setEnabled(false);
     }
 }
 
 void MainWindow::openLastSession()
 {
-    QFile description(m_session + ".johnny");
-    if (!description.open(QIODevice::ReadOnly | QIODevice::Text)) {
-        QMessageBox::critical(
-            this,
-            tr("Johnny"),
-            tr("Johnny could not open file to read session description!"));
-        return;
-    }
-    QTextStream descriptionStream(&description);
-    QString format;
-    QStringList fileNames;
-    // Parse johnny session file
-    while(!descriptionStream.atEnd())
-    {
-        QString content =  descriptionStream.readLine();
-        if(content.startsWith("FILE="))
-        {
-            content.remove(0,5);
-            fileNames.append(content);
-        }
-        else if(content.startsWith("FORMAT="))
-        {
-            content.remove(0,7);
-            format = content;
-        }
-    }
+    QString sessionName = QString(m_session).remove(m_appDataPath);
```

Comment:
> Leaner code:
> QString sessionName(m_session);
> m_session.remove(m_appDataPath);

Did you mean this ?
QString sessionName(m_session);
sessionName.remove(m_appDataPath);